### ISSUE TITLE
Build issue while pinning

### ISSUE THREAD

**Issue by liam923** [neuppl/rsdd-ocaml#5]
Following the instructions in the README, `opam pin rsdd ../path/to/rsdd-ocaml` produced the following error:

```
[ERROR] The compilation of rsdd.0.1.0 failed at "dune build -p rsdd -j 7 @install".

#=== ERROR while compiling rsdd.0.1.0 =========================================#
# context     2.1.4 | macos/x86_64 | ocaml-base-compiler.4.14.1 | pinned(git+file:///Users/liamstevenson/Projects/cs7470/rsdd-ocaml#main#5b81c32f)
# path        ~/Projects/cs7470/disc/_opam/.opam-switch/build/rsdd.0.1.0
# command     ~/.opam/opam-init/hooks/sandbox.sh build dune build -p rsdd -j 7 @install
# exit-code   1
# env-file    ~/.opam/log/rsdd-78569-ec051a.env
# output-file ~/.opam/log/rsdd-78569-ec051a.out
### output ###
# error: failed to get `rsdd` as a dependency of package `rsdd v0.1.0 (/Users/liamstevenson/Projects/cs7470/disc/_opam/.opam-switch/build/rsdd.0.1.0)`
# [...]
# Caused by:
#   failed to load source for dependency `rsdd`
# 
# Caused by:
#   Unable to update https://github.com/neuppl/rsdd?rev=c477465#c477465d
# 
# Caused by:
#   failed to create directory `/Users/liamstevenson/.cargo/git/db/rsdd-70c708972b058342`
# 
# Caused by:
#   Operation not permitted (os error 1)
```

I was able to resolve this by running `dune build` from inside the `rsdd-ocaml` directory (using the same opam switch, unsure if that's important) and then running `opam pin rsdd ../path/to/rsdd-ocaml` again. Presumably, `dune build` from inside the `rsdd-ocaml` directory successfully cloned `https://github.com/neuppl/rsdd?rev=c477465#c477465d` to `/Users/liamstevenson/.cargo/git/db/rsdd-70c708972b058342`, preventing the error from happening.

Of note: `/Users/liamstevenson/.cargo/git` did not exist before or after initially running the pin command. I forgot to check if it got created after running `dune build` / before running pin the second time.

Opam version: 2.1.4
OCaml version: 4.14.1
Dune version: 3.9.1
OS: MacOS Ventura 13.4.1
Rust/Cargo version: 1.65.0

**Comment by mattxwang** [comment]
Thanks for the detailed error report @liam923! I realized after some digging that the `.cargo/config` we bundled was out-of-date with the RSDD version set vendored by `cargo vendor`. I updated both in lockstep in [9749c5e](https://github.com/neuppl/rsdd-ocaml/commit/9749c5e419637ec711a7954b8b50bf4038998a98), which I *think* should resolve this issue; I confirmed that this doesn't happen anymore on [the opam CI runners](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/92733d2634c3c5cfa0fc44b03553d3d8998f2111).

There's no rush, but if you have a chance to try this locally from scratch again, that would be great! I'll leave this issue up for a bit in case any other students run into this problem.
